VmAgent.Core/Interfaces/DockerContainerEngine.cs

```diff
@@ -415,22 +415,14 @@
             return sessionHost;
         }
 
-        public static string GetImageNameFromContainerImageDetails(ContainerImageDetails imageDetails)
+        private static string GetImageNameFromContainerImageDetails(ContainerImageDetails imageDetails)
         {
             if (imageDetails == null)
             {
                 throw new ArgumentNullException(nameof(imageDetails));
             }
 
-            string imageName;
-            if (!string.IsNullOrEmpty(imageDetails.ImageDigest))
-            {
-                imageName = $"{imageDetails.ImageName}@{imageDetails.ImageDigest}";
-            }
-            else
-            {
-                imageName = $"{imageDetails.ImageName}:{imageDetails.ImageTag ?? "latest"}";
-            }
+            string imageName = imageDetails.ImageName;
 
             // Support running local images with no explicit registry in the name.
             if (!string.IsNullOrEmpty(imageDetails.Registry))
```

Comment:
IIRC, this was to support "local" images being built locally and consumed locally by LMA. Does this work now with this change?

Reply:
Short answer: yes.

Long answer:
When pulling images from the remote registry, prior to this change we assumed that the registry was specified and always appended it to the image name (so if the registry is null we would attempt to pull "<null>/myImage:tag"). When creating containers locally, we had logic to only append the registry if it was specified put in place for LocalMultiplayerAgent (so if the registry is null we attempt to create a container using image "myImage:tag"). Now with my change, since the logic is unified in both places we will attempt to use image "myImage:tag" instead of "<null>/myImage:tag" if the registry is null.

Reply:
Thanks. Let's do a LocalMultiplayerAgent release when we merge this, just to make sure!